frontend/lib/posting_view/model/posting.model.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:convert';
+
+PostingModel postingModelFromJson(String str, String itemid) =>
+    PostingModel.fromJson(json.decode(str), itemid);
+
+String postingModelToJson(PostingModel data) => json.encode(data.toJson());
+
+class PostingModel {
+  PostingModel({
+    Item? item,
+  }) : item = item ?? Item(name: '', price: 0.0);
+
+  Item item;
+
+  factory PostingModel.fromJson(Map<String, dynamic> json, String itemid) {
+    List<Item> items =
+        List<Item>.from(json["items"].map((x) => Item.fromJson(x)));
+    Item? foundItem = items.firstWhere((item) => item.id == itemid,
```

Comment:
Likely unnecessary but leave this to be fixed when the backend is complete for this page. It sort of depends on the kind of data you will receive.